### ISSUE TITLE
Build with Android O SDK (Android 8.0 / API 26)

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/focus-android#863]
This is at least needed to support home screen shortcuts on Android O (#833).

I tried using the new SDK in June (#758). Building worked fine but running our Robolectric tests failed.

* [x] Update SDK and other dependencies like support library in build.gradle files
* [x] Use new Shortcut API to implement #833
* [x] Use startForegroundService() to show permanent notification (this depends on multitasking refactoring)
* [x] Create a notification channel for our on-going notification
* [x] Fix other behavior changes or deprecated code: https://developer.android.com/preview/behavior-changes.html
* [x] Fix automated tests that depend on a different SDK version
* [x] Replace "placeholder code" that doesn't work with newest build tools anymore
* [x] Finalize notification channel strings (See comment with screenshots below)

**Comment by ekager** [comment]
Would we also have to refactor now-deprecated methods? If this isn't too complicated I think it would be great to be able to include homescreen shortcuts (#833) for Android O+ with the ShortcutManager. 👍

**Comment by pocmo** [comment]
If it's straight-forward to do. Otherwise I'd be okay to do it later. I'm more worried about some behavior changes in the SDK that break some of the existing features (like the ongoing notification). I'll try to get patches up for that soon! :)

**Comment by pocmo** [comment]
@antlam @mheubusch While Focus is active we show an on-going notification. Now on Android 8+ we need to create a so called "notification channel" for all notifications. Whenever we display a notification we need to assign it to a channel. A user can configure how (or if) notifications from a specific channel show up. Here's a pretty good article about the feature:
https://www.androidcentral.com/notification-channels

For our on-going notification we need to create a channel with a "name" and an (optional) description. The Android guidelines say:
* name: The recommended maximum length is 40 characters; the value may be truncated if it is too long.
* description: The recommended maximum length is 300 characters; the value may be truncated if it is too long.

I have absolutely no clue how we should name this thing.

However here are some screenshots from Android's UI showing the strings:

Showing all notification channels of the app. "Session" is how I named the channel in my build:
![overview](https://user-images.githubusercontent.com/89638/29579878-7571b628-8774-11e7-96f8-3a55df304828.png)

This is the screen shown when tapping on the channel to configure it. "Text text" is the description of the channel I used in my build:
![channel](https://user-images.githubusercontent.com/89638/29579925-9473b684-8774-11e7-86cf-8c1884628689.png)

And just in case, this is the notification we show:
![notification](https://user-images.githubusercontent.com/89638/29580034-e93ab226-8774-11e7-9be0-8836be1a93f4.png)
